### ISSUE TITLE
Windows 32 bits compilation error

### ISSUE THREAD

**Issue by gaaned92** [Beep6581/RawTherapee#4899]
When compiling last dev branch for Windows 32 bit (so without SSE2), I get these errors:

```
In file included from D:/RAWTHERAPEE/RTSOURCE/rawtherapee/rtengine/gauss.cc:23:0:
D:/RAWTHERAPEE/RTSOURCE/rawtherapee/rtengine/boxblur.h: In function 'void rtengine::boxblur(T**, A**, int, int, int, int)':
D:/RAWTHERAPEE/RTSOURCE/rawtherapee/rtengine/boxblur.h:38:5: error: there are no arguments to 'assert' that depend on a template parameter, so a declaration of'assert' must be available [-fpermissive]
     assert(2*radx+1 < W);
     ^~~~~~
D:/RAWTHERAPEE/RTSOURCE/rawtherapee/rtengine/boxblur.h:38:5: note: (if you use '-fpermissive', G++ will accept your code, but allowing the use of an undeclaredname is deprecated)
D:/RAWTHERAPEE/RTSOURCE/rawtherapee/rtengine/boxblur.h:39:5: error: there are no arguments to 'assert' that depend on a template parameter, so a declaration of'assert' must be available [-fpermissive]
     assert(2*rady+1 < H);
     ^~~~~~
```

**Comment by agriggio** [comment]
my bad, sorry. Just add `#include <assert.h>` at the beginning of the file

**Comment by ff2000** [comment]
I know assert.h usually works but the C headers in C++ got their own compatibility headers, here `<cassert>`. No idea how strict you are about this.

**Comment by Floessie** [comment]
@ff2000 You're absolutely right.

Those `<c...>` headers usually mean that the C functions will be inside namespace `std`, but this is not the case for the `assert` macro. Thus there's no `std::assert()` and including `<assert.h>` is just fine, IMHO.

But sure, this is a special case.

HTH,
Flössie

**Comment by agriggio** [comment]
I'm personally not very strict about this. in my daily activities, I mix c and c++ fairly often (either directly or via libraries), so I just don't see the point of the c-prefixed headers. but if you want to use that, no problem

**Comment by gaaned92** [comment]
@agriggio 
`Just add #include <assert.h> at the beginning of the file`
It builds ok in W32.
Could you commit the modification in the dev branch please?

**Comment by heckflosse** [comment]
Fixed with 66bb7ba
